test/widgets/action_sheet_test.dart

```diff
@@ -11,13 +11,18 @@ import 'package:zulip/widgets/compose_box.dart';
 import 'package:zulip/widgets/content.dart';
 import 'package:zulip/widgets/message_list.dart';
 import 'package:zulip/widgets/store.dart';
+// We want whatever version of `share_plus_platform_interface` is pulled in by
+// `share_plus`; we don't want to pull in some other version ourselves.
+// ignore: depend_on_referenced_packages
+import 'package:share_plus_platform_interface/method_channel/method_channel_share.dart';
```

Comment:
That's a concern with NPM and related package managers in the JS world, but I believe it's not possible with Dart's `pub` package manager. The latter's architecture is that a given package is installed just once for a given app, not potentially several times for different sub-dependencies.

You can see this reflected in the structure of `pubspeck.lock` — the keys under `packages` are just the package names, so there's at most one entry per package. In `yarn.lock`, by contrast, the keys are package names-plus-versions, which lets there be several for a given package.

This is also why the `test_api` package can be used to ensure `flutter_test` and `test` are at compatible versions just by having the latter each point to it as a normal dependency:
```
$ grep -B1 test_api ~/.pub-cache/hosted/pub.dev/test-1.24.9/pubspec.yaml 

  # Use an exact version until the test_api and test_core package are stable.
  test_api: 0.6.1
```

without any extra mechanisms like NPM's "peer dependencies".

So we should just do what the lint says here — any version of `share_plus_platform_interface` we get will necessarily be the same one that we're providing to `share_plus`.

Reply:
Interesting. Thanks for explaining!